lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -470,6 +478,7 @@ def _call(self, method: Callable, json_data: Dict[AnyStr, Any]):
             )
             abort(500, message="INTERNAL ERROR")
 
+    @Auth.token_auth.login_required()
```

Comment:
Authorization has to be _optional_ because a client without a login needs to be able to query for public data.

You could further finesse this by requiring `"user": None` in this case, and defaulting a missing `"user"` parameter in the JSON to `None`. (Note, that's why I referenced the subclass `assembly` methods in the email.)

This goes for the `get` as well.

Reply:
Yeah I will make the authentication optional

Reply:
Okay I think it would be better if we can somehow separate the public data queries from the queries that does need a user authentication, that way we dont have to manage how the "user" parameter is formatted in the json request in order to understand if it is a public dataset query or an actual user dataset query. For example we can have an endpoint `/controllers/list` that requires a "user" in the json request and an authorization header that belongs to the user in the query. We can also have an endpoint `/controllers/list/public` that does not accept the "user" parameter in the json request, only "start" and "end" and this endpoint does not require an authorization header but internally we do the exact same thing as a class `ControllersList` with user as `None`

This way I feel its more cleaner and probably less prone to security vulnerability (we wont proceed to query the request if the endpoint is strictly protected rather than allowing the optional authorization header). And also maybe it will be easirt to adopt when we have role based authentication

Reply:
The endpoint shouldn't change, _especially_ based on roles, because the client shouldn't have to know the details of the security token it holds; the server takes care of that. The client's visibility to data changes, but not how they (attempt to) access it. I really don't like this idea of fragmenting APIs; "that way lies madness". (Well, at least a major pain in the butt.)

The authorization token tells the server ("out of band", essentially) what you're _allowed_ to access; the `"user"` parameter tells it what you _want_ to access. If the two scopes overlap, you get what you want; if not, you get a security violation. As I said, we can be "smart" and default a missing `"user"` based on the actual security domain (`"user": ""` or `"user": None` for missing authorization token, or `"user": "<logged-in-user>"` with a token) because that's a convenient and meaningful default... essentially we don't make them specify `"user"` unless they want something unusual, which gets into privilege and roles).

Reply:
Fixed